### PR TITLE
Add fromUrl method

### LINKED ISSUE #705: Send media with sendMessage by media URL

### LINKED ISSUE #771: cannot send media
Hello,

I just tried your api, this could be send message as well (according on example.js), but how to send media, because when I tried to send media like this : 

```
client.on('ready', () => {
	const media = new MessageMedia(
        'image/png', 
        'https://homepages.cae.wisc.edu/~ece533/images/boat.png'
    );

    client.sendMessage("614xxxxxxxxx@c.us",media).then((r) => {
        console.log("sendMessage",r)
    })
});
```

it always return error :  _Evaluation failed: DOMException: Failed to execute 'atob' on 'Window': The string to be decoded is not correctly encoded._

Please help me

Thanks

### PR DESCRIPTION
**Pull request by wictornogueira**
Does basically the same thing as fromFilePath but takes a url instead

### CODE REVIEW

**Review by Gimenz — APPROVED** [review]

**Comment by PurpShell** [comment]
Couldn't we use `http` instead of `node-fetch` here? Also you haven't updated `package.json`

**Comment by wictornogueira** [comment]
> Couldn't we use `http` instead of `node-fetch` here? Also you haven't updated `package.json`
> 
> also why are you wrapping the entire function in a `Promise`? Wouldn't most of that job be done if you used `async` and `await fetch()` instead of `fetch().then`
> 
> Also you are determining the mime-type via the URL itself, instead of using `mime`, which i think is less relevant in today's standards, maybe get the data via `node-fetch`'s response "content-type" header

1. I would have to write a wrapper in order to use http, which would be a bit overkill
2. I haven't added any dependency to the dependency tree; `node-fetch` is a depency of `puppeteer`
3. Using Promise makes things easir, beucase It allows me to easly resolve it by calling resolve
4. Promises also allow you to raise an exception from anywhere inside its scope
5.
  5.1. Some server still respond with `application/octet-stream`, which is not exactly ideal
  5.2 Altho I've specified the `Content-type` I would accept, it's up to the server to respect it or not
  5.3 Not even browsers trust `Content-type`
  5.4 Saves some time and bandwidth by only downloading something that is more likely to work
  5.5 And I know all this comes with the tradeo-off of not being able to use weird urls

**Comment by Gimenz** [comment]
how about request config options?

**Comment by wictornogueira** [comment]
> how about request config options?

What do you mean?

**Comment by Gimenz** [comment]
> > how about request config options?
> 
> What do you mean?

request options..
like headers, user-agent, cookie.. etc

**Comment by wictornogueira** [comment]
> > > how about request config options?
> > 
> > 
> > What do you mean?
> 
> request options..
> like headers, user-agent, cookie.. etc

How does it help?

**Comment by Gimenz** [comment]
> > > > how about request config options?
> > > 
> > > 
> > > What do you mean?
> > 
> > 
> > request options..
> > like headers, user-agent, cookie.. etc
> 
> How does it help?

maybe, it will help if some cases need headers

for example :

```javascript
MessageMedia.fromUrl('url', 0, { 
    headers: {
        Cookie: 'some cookie',
        'User-Agent': 'ua'
    }
})
```

like this [getDUrl](https://github.com/open-wa/wa-automate-nodejs/blob/5b7f2406f7d52a538ffd85c3513a24c666103d45/src/utils/tools.ts#L63)

**Comment by wictornogueira** [comment]
Ohhh... yeah, indeed.

File: src/structures/MessageMedia.js [hunk 1]
```diff
@@ -43,6 +45,35 @@ class MessageMedia {
...
+     * @param {number} [sizeLimit=0]
+     * @returns {Promise<MessageMedia>}
+     */
+    static fromUrl(url, sizeLimit = 0) {
+        return new Promise((resolve, reject) => {
+            const pUrl = new URL(url);
+            const mimetype = mime.getType(pUrl.pathname);
+
+            if (!mimetype) {
+                return reject(new Error('Unable to determine MIME type'));
```
Comment (tuyuribr) [inline src/structures/MessageMedia.js:61]: I think that rejecting when you are unable to get he mime type would be an issue, how about a param "ignoreMime" ?

**Comment by PurpShell** [comment]
> Ohhh... yeah, indeed.

yeah also some users are using a proxy keep that in mind

**Comment by wictornogueira** [comment]
You can now skip the initial mime checking thing, use proxy, override request options and choose context

**Review by purpshell — APPROVED** [review]
LGTM

**Review by aliyss — APPROVED** [review]
lgtm

**Comment by iriobf** [comment]
How to set the filename in the fromUrl method?
Media being sent with filename null.
```javascript
const media = await MessageMedia.fromUrl(fileUrl)
const msgMedia = await sessions[from].sendMessage(numberTo, media)
```

**Comment by wictornogueira** [comment]
```js
media.filename = 'whatever'
```

**Comment by d0v3riz** [comment]
> ```js
> media.filename = 'whatever'
> ```

Can I get the original filename from that url?

**Comment by iriobf** [comment]
> > ```js
> > media.filename = 'whatever'
> > ```
> 
> Can I get the original filename from that url?

media.filename = new URL(url).pathname.split('/').pop();

**Pull request merged**